bindgen/lib.rs

```diff
@@ -1232,9 +1241,50 @@ fn get_target_dependent_env_var(
 ///     .generate();
 /// ```
 #[derive(Debug)]
-pub struct CargoCallbacks;
+pub struct CargoCallbacks {
+    rerun_on_input_files: bool,
+}
+
+/// Create a new `CargoCallbacks` value with [`CargoCallbacks::rerun_on_input_files`] disabled.
+///
+/// This constructor has been deprecated in favor of [`CargoCallbacks::new`] where
+/// [`CargoCallbacks::rerun_on_input_files`] is enabled by default.
+#[deprecated = "Use `CargoCallbacks::new()` instead. Please, check the documentation for further information."]
+pub const CargoCallbacks: CargoCallbacks = CargoCallbacks {
+    rerun_on_input_files: false,
+};
+
+impl CargoCallbacks {
+    /// Create a new `CargoCallbacks` value.
+    pub fn new() -> Self {
```

Comment:
Could make this const, and use `new` in the deprecated const as well?

Reply:
I'm a bit conservative with `const fn`s as removing the `const` in the future will always be a breaking change and I'm not sure new fields will be added that won't be able to be initialized using constants in the future.